pkg/server/service/tcp/service.go

```diff
@@ -91,14 +96,33 @@ func (m *Manager) BuildTCP(rootCtx context.Context, serviceName string) (tcp.Han
 				continue
 			}
 
+			tcpAddr, err := net.ResolveTCPAddr("tcp", server.Address)
+			if err != nil {
+				srvLogger.Error().Err(err).Msg("Failed to resolve TCP address")
+				continue
+			}
+			healthCheckTargets[server.Address] = tcpAddr
+
 			loadBalancer.AddServer(handler)
 			logger.Debug().Msg("Creating TCP server")
 		}
 
+		if conf.LoadBalancer.HealthCheck != nil {
+			m.healthCheckers[serviceName] = healthcheck.NewServiceTCPHealthChecker(
```

Comment:
Unless I'm wrong, the TCP health checkers are created but never launched. While HTTP health checkers are started via `serviceManager.LaunchHealthCheck(ctx)` in `pkg/server/routerfactory.go`, there's no equivalent call for TCP services.